sbc_email/models/import_letters_history.py

```diff
@@ -0,0 +1,384 @@
+                smb_conn.close()
+            else:
+                logger.info("""Failed to list files in imported \
+                folder Imports oh the NAS in emplacement: {}""".format(
+                    imported_letter_path))
+
+            self.nber_letters = tmp
+
+    ##########################################################################
+    #                             VIEW CALLBACKS                             #
+    ##########################################################################
+    @api.multi
+    def button_import(self):
+        """
+        Analyze the imports in order to create the letter's lines
+        """
+        if not self.manual_import:
+            # when letters are in a folder on NAS redefine method
+            for letters_import in self:
+                letters_import.state = 'pending'
+                if self.env.context.get('async_mode', True):
+                    session = ConnectorSession.from_env(self.env)
+                    ilh.import_letters_job.delay(
+                        session, self._name, letters_import.id)
+                else:
+                    letters_import._run_analyze()
+            return True
+        else:
+            # when letters selected by user, save them on NAS and call
+            # super method
+            for letters_import in self:
+                if letters_import.data and self.env.context.get(
+                        'async_mode', True):
+                    for attachment in letters_import.data:
+                        self._save_imported_letter(attachment)
+
+            return super(ImportLettersHistory, self).button_import()
+
+    #########################################################################
+    #                             PRIVATE METHODS                           #
+    #########################################################################
+    def _run_analyze(self):
+        """
+        Analyze each attachment:
+        - check for duplicate file names and skip them
+        - decompress zip file if necessary
+        - call _analyze_attachment for every resulting file
+        """
+        self.ensure_one()
+        # keep track of file names to detect duplicates
+        # file_name_history = []
+        logger.info("Imported letters analysis started...")
+        progress = 1
+
+        # Case where letters are in 'Imports' folder on the NAS
+        config_obj = self.env['ir.config_parameter']
+        share_nas = (config_obj.search(
+            [('key', '=', 'sbc_email.share_on_nas')])[0]).value
+
+        # Retrieve SMB configuration
+        smb_user = config.get('smb_user')
+        smb_pass = config.get('smb_pwd')
+        smb_ip = config.get('smb_ip')
+        smb_port = int(config.get('smb_port', 0))
+        if not (smb_user and smb_pass and smb_ip and smb_port):
+            return False
+        smb_conn = SMBConnection(smb_user, smb_pass, 'openerp', 'nas')
```

Comment:
You could put your connection code inside a method since you use it twice in the class (to avoid duplicate code).